src/Npgsql/TypeHandlers/HstoreHandler.cs

```diff
@@ -182,6 +187,19 @@ async ValueTask<T> ReadInto<T>(T dictionary, int numElements, NpgsqlReadBuffer b
 
         #endregion
 #endif
+
+        #region IReadOnlyDictionary
+
+        int INpgsqlTypeHandler<IReadOnlyDictionary<string, string?>>.ValidateAndGetLength(IReadOnlyDictionary<string, string?> value, ref NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter)
+            => ValidateAndGetLength(value, ref lengthCache, parameter);
+
+        Task INpgsqlTypeHandler<IReadOnlyDictionary<string, string?>>.Write(IReadOnlyDictionary<string, string?> value, NpgsqlWriteBuffer buf, NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter, bool async)
+            => Write(value, value.Count, buf, lengthCache, parameter, async);
+
+        ValueTask<IReadOnlyDictionary<string, string?>> INpgsqlTypeHandler<IReadOnlyDictionary<string, string?>>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription)
+            => new ValueTask<IReadOnlyDictionary<string, string?>>(Read(buf, len, async, fieldDescription).Result);
```

Comment:
Getting result of an incomplete `ValueTask<T>` may cause an exception if `IValueTaskSource` is used under the hood. In case of `Task` this method will block the current thread. Therefore, you need to rewrite this as:

```csharp
        async ValueTask<IReadOnlyDictionary<string, string?>> INpgsqlTypeHandler<IReadOnlyDictionary<string, string?>>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription)
            => await Read(buf, len, async, fieldDescription);
```